### PR TITLE
fix(sdk): Update wait_for_run_completion function to handle tekton status;

### LINKED ISSUE #1223: Where is the source code for wait_for_run_completion?
/kind question

**Question:**
Where is the source code for `wait_for_run_completion`?

**Description:**
Hello,

I am working on an integration test. I have the following:
```
print(kfp_tekton.__version__)
'1.5.3'
```
My code is doing the following
```
my_pipeline = ....
result = client.create_run_from_pipeline_func(pipeline_func=my_pipeline, arguments={})
client.wait_for_run_completion(result.run_id, 2 * 60)
```

I am looking at: https://github.com/kubeflow/kfp-tekton/blob/1.5.x/sdk/python/kfp_tekton/_client.py
I can't see the method. Where is it?

Thanks,

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #1223 

**Description of your changes:**

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp_tekton/_client.py [hunk 1]
```diff
@@ -159,6 +161,61 @@ def __init__(self,
...
+                    continue
+                else:
+                    raise api_ex
+            status = get_run_response.run.status
+            elapsed_time = (datetime.datetime.now() -
+                            start_time).total_seconds()
+            logging.info('Waiting for the job to complete...')
+            if elapsed_time > timeout:
+                raise TimeoutError('Run timeout')
+            time.sleep(5)
```
Comment (yhwang) [inline sdk/python/kfp_tekton/_client.py:217]: nit: even if the pipeline is done, you still need to wait for 5 seconds

Comment (Tomcli) [reply sdk/python/kfp_tekton/_client.py:217]: This is from upstream. Let me fix it here and summit a pr upstream as well.

Comment (Tomcli) [reply sdk/python/kfp_tekton/_client.py:219]: I optimized it better so it returns the status if completed without waiting

**Review by yhwang — APPROVED** [review]
/lgtm

**Comment by Tomcli** [comment]
> Will it be backported to `1.5.x` Will we have a new version of `1.5.4`?

Yes, we just need some time to cherrypick all the changes back to 1.5.x branch and run our release ci.

**Pull request merged**